llmfoundry/data/text_data.py

```diff
@@ -293,6 +295,47 @@ def build_text_dataloader(
         timeout=cfg.get('timeout', 0),
     )
 
+    # If we pretokenized, we may not have padding, in which the
+    # tokenizer may not have a pad_token_id. In this case, we can
+    # just use the default token counting function.
+    token_counting_func = None
+    if tokenizer.pad_token_id is not None:
```

Comment:
This may lead to inaccurate counts if the tokenizer config during pretokenization doesn't match the tokenizer at this point.

Reply:
That is true. I believe we don't support training in this fashion (that is, if you put "padding" tokens into your pretokenized data, they will simply be trained on like normal tokens. But I will leave a note behind here with a bit more explanation.